core-backend/sandbox/src/funcs.rs

```diff
@@ -623,22 +623,26 @@ pub fn create_program_wgas<E: Ext + Into<ExtInfo>>(
 ) -> Result<ReturnValue, HostError> {
     let mut args = args.iter();
 
-    let code_hash_ptr = pop_i32(&mut args)?;
+    let fl_ptr = pop_i32(&mut args)?;
     let salt_ptr = pop_i32(&mut args)?;
     let salt_len = pop_i32(&mut args)?;
     let payload_ptr = pop_i32(&mut args)?;
     let payload_len = pop_i32(&mut args)?;
-    let gas_limit = pop_i64(&mut args)?;
-    let value_ptr = pop_i32(&mut args)?;
     let program_id_ptr = pop_i32(&mut args)?;
 
     let result = ctx
         .ext
         .with(|ext: &mut E| -> Result<(), &'static str> {
-            let code_hash = funcs::get_bytes32(ext, code_hash_ptr);
+            // Handling data under `fl_ptr`, which is a pointer to multiple fixed len params needed to create a [`ProgramInitPacket`].
+            // For more info see [`gcore::prog::sys::gr_create_program_wgas`].
+            let code_hash = funcs::get_fixed_bytes(ext, fl_ptr);
+            let gas_limit = funcs::get_u64(ext, fl_ptr + code_hash.len());
+            let value = funcs::get_u128(
+                ext,
+                fl_ptr + code_hash.len() + gas_limit.to_be_bytes().len(),
```

Comment:
Why big-endian and not [size_of_val()](https://doc.rust-lang.org/core/mem/fn.size_of_val.html)?

```suggestion
                fl_ptr + code_hash.len() + core::mem::size_of_val(&gas_limit),
```